api/stacks/deployments/compose_unpacker_cmd_builder.go

```diff
@@ -0,0 +1,220 @@
+package deployments
+
+import (
+	"fmt"
+
+	portainer "github.com/portainer/portainer/api"
+	"github.com/portainer/portainer/api/dataservices"
+	"github.com/portainer/portainer/api/internal/registryutils"
+)
+
+type StackRemoteOperation string
+
+const (
+	OperationDeploy        StackRemoteOperation = "compose-deploy"
+	OperationUndeploy      StackRemoteOperation = "compose-undeploy"
+	OperationComposeStart  StackRemoteOperation = "compose-start"
+	OperationComposeStop   StackRemoteOperation = "compose-stop"
+	OperationSwarmDeploy   StackRemoteOperation = "swarm-deploy"
+	OperationSwarmUndeploy StackRemoteOperation = "swarm-undeploy"
+	OperationSwarmStart    StackRemoteOperation = "swarm-start"
+	OperationSwarmStop     StackRemoteOperation = "swarm-stop"
+)
+
+const (
+	UnpackerCmdDeploy        = "deploy"
+	UnpackerCmdUndeploy      = "undeploy"
+	UnpackerCmdSwarmDeploy   = "swarm-deploy"
+	UnpackerCmdSwarmUndeploy = "swarm-undeploy"
+)
+
+type unpackerCmdBuilderOptions struct {
+	stackOperation     StackRemoteOperation
+	pullImage          bool
+	prune              bool
+	composeDestination string
+	registries         []portainer.Registry
+}
+
+type BuildCmdFunc func(stack *portainer.Stack, opts unpackerCmdBuilderOptions, registries []string, env []string) []string
+
+// build the unpacker cmd for stack based on stackOperation
+func (d *stackDeployer) buildUnpackerCmdForStack(stack *portainer.Stack, opts unpackerCmdBuilderOptions) []string {
+
+	funcmap := map[StackRemoteOperation]BuildCmdFunc{
+		OperationDeploy:        buildDeployCmd,
+		OperationUndeploy:      buildUndeployCmd,
+		OperationComposeStart:  buildComposeStartCmd,
+		OperationComposeStop:   buildComposeStopCmd,
+		OperationSwarmDeploy:   buildSwarmDeployCmd,
+		OperationSwarmUndeploy: buildSwarmUndeployCmd,
+		OperationSwarmStart:    buildSwarmStartCmd,
+		OperationSwarmStop:     buildSwarmStopCmd,
+	}
+
+	registriesStrings := getRegistry(opts.registries, d.dataStore)
+	envStrings := getEnv(stack.Env)
+
+	return funcmap[opts.stackOperation](stack, opts, registriesStrings, envStrings)
```

Comment:
may I suggest moving `stackOpertion` to be a parameter of the function, because it's required (and we don't need it for the next calls).

also, do we need to check if `opts.stackOperation` is a function?